ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -541,32 +549,48 @@ def run_ocs_upgrade(operation=None, *operation_args, **operation_kwargs):
             upgrade=True
         )
         log.info(f"Disconnected upgrade - new image: {upgrade_ocs.ocs_registry_image}")
+
     with CephHealthMonitor(ceph_cluster):
         channel = upgrade_ocs.set_upgrade_channel()
         upgrade_ocs.set_upgrade_images()
-        if upgrade_version != "4.9":
-            # In the case of upgrade to ODF 4.9, the ODF operator should upgrade
-            # OCS automatically.
-            upgrade_ocs.update_subscription(channel)
-        if original_ocs_version == "4.8" and upgrade_version == "4.9":
-            deployment = Deployment()
-            deployment.subscribe_ocs()
-        if (config.ENV_DATA["platform"] == constants.IBMCLOUD_PLATFORM) and not (
-            upgrade_in_current_source
-        ):
-            create_ocs_secret(config.ENV_DATA["cluster_namespace"])
-            for attempt in range(2):
-                # We need to do it twice, because some of the SA are updated
-                # after the first load of OCS pod after upgrade. So we need to
-                # link updated SA again.
-                log.info(
-                    f"Sleep 1 minute before attempt: {attempt+1}/2 "
-                    "of linking secret/SAs"
-                )
-                time.sleep(60)
-                link_all_sa_and_secret_and_delete_pods(
-                    constants.OCS_SECRET, config.ENV_DATA["cluster_namespace"]
+        ui_upgrade_supported = False
+        if config.UPGRADE.get("ui_upgrade"):
+            if (
+                version.get_semantic_ocp_version_from_config() == version.VERSION_4_9
+                and original_ocs_version == "4.8"
+                and upgrade_version == "4.9"
+            ):
+                ui_upgrade_supported = True
+            else:
+                log.warning(
+                    "UI upgrade combination is not supported. It will fallback to CLI upgrade"
                 )
+            if ui_upgrade_supported:
+                ocs_odf_upgrade_ui()
+        else:
+            if upgrade_version != "4.9":
+                # In the case of upgrade to ODF 4.9, the ODF operator should upgrade
```

Comment:
won't it be the case also for upgrade_version > 4.9?

Reply:
discussed offline and will be taken care of by a separate PR

Reply:
I was thinking that this condition is already meeting the criteria **upgrade_version > 4.9** because the code says , if **upgrade_version != "4.9":**, so not equal to means lesser or greater both. Or Am I wrong?

Reply:
As discussed offline, this looks fine. :)